src/checkpoint/container/view_serialize.h

```diff
@@ -264,12 +264,19 @@ inline void serialize_impl(SerializerT& s, Kokkos::DynRankView<T,Args...>& view)
   // Serialize the Kokkos layout data, including the extents, strides
   ArrayLayoutType layout;
 
-  // Make sure we serialize all 7 dimensions, instead of just `dims`. The
-  // other dimensions contain a 1 to make sure the size comes out correctly
+  // Make sure we serialize all 7 dimensions, instead of just `dims`.
   if (s.isUnpacking()) {
     serializeLayout<SerializerT>(s, 7, layout);
   } else {
     ArrayLayoutType layout_cur = view.layout();
+
+    // We must set these to the invalid index (they are not set in the layout by
+    // default!). This ensures that when the DynRankView comes out after
+    // de-serialization, the number of ranks is correct.
+    for (int i = dims; i < 8; i++) {
```

Comment:
I just realized - should this loop bound be 8, or 7? Kokkos Views allow up to 7 dimensions. Is there some weird 1-based indexing of `ArrayLayout`?

Reply:
Never mind, looks like 8 is right
https://github.com/kokkos/kokkos/wiki/Kokkos%3A%3ALayoutRight

Reply:
Ugh, documentation says View goes up to 8 dimensions, while DynRankView goes to 7
https://github.com/kokkos/kokkos/wiki/View
https://github.com/kokkos/kokkos/wiki/Kokkos%3A%3ADynRankView